includes/class-amp-wp-styles.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/**
+ * Class AMP_WP_Styles
+ *
+ * @package AMP
+ */
+
+/**
+ * Extend WP_Styles with handling of CSS for AMP.
+ *
+ * @since 0.7
+ */
+class AMP_WP_Styles extends WP_Styles {
+
+	/**
+	 * Concatenation is always enabled for AMP.
+	 *
+	 * @since 0.7
+	 * @var bool
+	 */
+	public $do_concat = true;
+
+	/**
+	 * Generates an enqueued style's fully-qualified file path.
+	 *
+	 * @since 0.7
+	 * @see WP_Styles::_css_href()
+	 *
+	 * @param string $src The source URL of the enqueued style.
+	 * @param string $handle The style's registered handle.
+	 * @return string|WP_Error Style's absolute validated filesystem path, or WP_Error when error.
+	 */
+	public function get_validated_css_file_path( $src, $handle ) {
+		if ( ! is_bool( $src ) && ! preg_match( '|^(https?:)?//|', $src ) && ! ( $this->content_url && 0 === strpos( $src, $this->content_url ) ) ) {
```

Comment:
It's like this because it is forked/copied from core: https://github.com/WordPress/wordpress-develop/blob/4fb6c02b9961d41dcaaf0ea04e24223993fa8667/src/wp-includes/class.wp-styles.php#L327